### PR TITLE
 MM-31717: Remove HTTP clustering

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://github.com/mattermost/mattermost-server/pull/17409

Updated:
- Self-Managed Admin Guide > Configure Mattermost > Configuration Settings > Environment > High Availability
   - Updated the Use Gossip configuration setting with a note indicating that Gossip clustering can no longer be disabled from Mattermost Server, and that all cluster traffic uses the gossip protocol.

### CODE REVIEW

File: source/administration/config-settings.rst [hunk 1]
```diff
@@ -1019,6 +1019,9 @@ Use IP Address
 Use Gossip
 ^^^^^^^^^^
 
+.. note::
+   Gossip clustering can no longer be disabled from Mattermost Server v5.35. All cluster traffic uses the gossip protocol.
```
Comment (justinegeffen) [inline source/administration/config-settings.rst:1023]:
```suggestion
   All cluster traffic uses the gossip protocol. From Mattermost Server v5.35 gossip clustering can no longer be disabled.
```

File: source/deployment/cluster.rst [hunk 2]
```diff
@@ -446,6 +446,13 @@ Starting with Mattermost Server version 4.0, when a server starts up it can auto
 8. When the server is running, start the other servers.
 9. Restart NGINX.
 
+All Cluster Nodes Must Use a Single Protocol
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+From Mattermost Server v5.35, `gossip clustering can no longer be disabled <https://docs.mattermost.com/administration/config-settings.html#use-gossip>`__. All cluster traffic uses the gossip protocol.
```
Comment (justinegeffen) [inline source/deployment/cluster.rst:452]:
```suggestion
All cluster traffic uses the gossip protocol. From Mattermost Server v5.35, `gossip clustering can no longer be disabled <https://docs.mattermost.com/administration/config-settings.html#use-gossip>`__.
```

File: source/deployment/cluster.rst [hunk 3]
```diff
@@ -446,6 +446,13 @@ Starting with Mattermost Server version 4.0, when a server starts up it can auto
...
+
+When upgrading a High Availability cluster, you can't upgrade other nodes in the cluster when one node isn't using the gossip protocol. You must use gossip to complete a High Availability upgrade, or you can shut down all nodes and bring them all up following an upgrade.
```
Comment (justinegeffen) [inline source/deployment/cluster.rst:454]:
```suggestion
When upgrading a High Availability cluster, you can't upgrade other nodes in the cluster when one node isn't using the gossip protocol. You must use gossip to complete a High Availability upgrade. Alternatively you can shut down all nodes and bring them all up individually following an upgrade.
```

**Review by justinegeffen — COMMENTED** [review]
I added two rewording suggestions. My third suggestion seemed like it may work but may also be missing context!

**Comment by cwarnermm** [comment]
Great updates, @justinegeffen -- Thank you!

**Review by justinegeffen — APPROVED** [review]

**Review by agnivade — CHANGES_REQUESTED** [review]
From @amyblais's comment [here](https://github.com/mattermost/mattermost-server/pull/17409#issuecomment-820370093), I was of the understanding that this is coming in 5.36, not 5.35. Can you please double-check? Thanks.

**Comment by cwarnermm** [comment]
Thanks so much @agnivade! You're absolutely right - this is a v5.36 update. I'll update the version references in the docs.

File: source/administration/config-settings.rst [hunk 4]
```diff
@@ -1019,6 +1019,9 @@ Use IP Address
 Use Gossip
 ^^^^^^^^^^
 
+.. note::
+   All cluster traffic uses the gossip protocol. From Mattermost Server v5.35 gossip clustering can no longer be disabled.
```
Comment (cwarnermm) [inline source/administration/config-settings.rst:1023]:
```suggestion
   All cluster traffic uses the gossip protocol. From Mattermost Server v5.36 gossip clustering can no longer be disabled.
```

File: source/deployment/cluster.rst [hunk 5]
```diff
@@ -446,6 +446,13 @@ Starting with Mattermost Server version 4.0, when a server starts up it can auto
 8. When the server is running, start the other servers.
 9. Restart NGINX.
 
+All Cluster Nodes Must Use a Single Protocol
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All cluster traffic uses the gossip protocol. From Mattermost Server v5.35, `gossip clustering can no longer be disabled <https://docs.mattermost.com/administration/config-settings.html#use-gossip>`__.
```
Comment (cwarnermm) [inline source/deployment/cluster.rst:452]:
```suggestion
All cluster traffic uses the gossip protocol. From Mattermost Server v5.36, `gossip clustering can no longer be disabled <https://docs.mattermost.com/administration/config-settings.html#use-gossip>`__.
```

**Review by wiersgallak — APPROVED** [review]

**Review by agnivade — APPROVED** [review]

**Pull request merged**